lib/orderbook/OrderBook.ts

```diff
@@ -135,6 +149,56 @@ class OrderBook extends EventEmitter {
     }
   }
 
+  public addPair = async (pair: Pair) => {
+    const pairId = derivePairId(pair);
+    if (this.pairs.has(pairId)) {
+      throw errors.PAIR_ALREADY_EXISTS(pairId);
+    }
+    if (!this.currencies.has(pair.baseCurrency)) {
+      throw errors.UNSUPPORTED_CURRENCY(pair.baseCurrency);
+    }
+    if (!this.currencies.has(pair.quoteCurrency)) {
+      throw errors.UNSUPPORTED_CURRENCY(pair.quoteCurrency);
+    }
+
+    const pairInstance = await this.repository.addPair(pair);
+    this.pairs.set(pairInstance.id, pairInstance);
+    return pairInstance;
+  }
+
+  public addCurrency = async (currency: CurrencyFactory) => {
+    if (this.currencies.has(currency.id)) {
+      throw errors.CURRENCY_ALREADY_EXISTS(currency.id);
+    }
+    const currencyInstance = await this.repository.addCurrency(currency);
+    this.currencies.set(currencyInstance.id, currencyInstance);
+  }
+
+  public removeCurrency = (currencyId: string) => {
+    const currency = this.currencies.get(currencyId);
+    if (currency) {
+      for (const pairId of this.pairIds) {
+        if (pairId.includes(currencyId)) {
+          throw errors.CURRENCY_CANNOT_BE_REMOVED(currencyId, pairId);
```

Comment:
If the currency is in multiple pairs the user has to call `removeCurrency` and `removePair` multiple times to know the blocking pairs and remove them. I think all affected pairs should be returned at once. But that can be done in a separate PR.

Btw: what is more efficient `pairId.includes(currencyId)` or checking if `currencyId` equals either `baseCurrency` or `quoteCurrency`?

Reply:
I was curious and it looks like the latter is indeed faster: http://jsben.ch/nIxYb

I'm sure it won't make a difference but I'll have it use the faster way.